src/Common/BlechString.fs

```diff
@@ -0,0 +1,490 @@
+
+
+    //let decimalToChar (decimal : int) =
+    //    assert (0 <= decimal && decimal <= 255)
+    //    string decimal
+
+    //let private decimalEscapeTo3DigitDecimalEscape str = 
+    //    decimalEscapeToUtf32 str
+    //    |> decimalTo3DigitDecimalEscape
+
+    //let private decimalEscapeToOctalEscape str =
+    //    decimalEscapeToUtf32 str
+    //    |> decimalToOctalEscape 
+
+    //let private decimalEscapeToHexEscape str =
+    //    decimalEscapeToUtf32 str
+    //    |> decimalToHexEscape 
+
+
+    //let private decimalEscapesToUnicodeDecimals str =
+    //    let mev = MatchEvaluator (fun m -> decimalEscapeTo3DigitDecimalEscape m.Value) 
+    //    Regex.Replace(str, DecimalEscape, mev)
+    
+    //let private decimalEscapesToOctalEscapes str =
+    //    let mev = MatchEvaluator (fun m -> decimalEscapeToOctalEscape m.Value) 
+    //    Regex.Replace(str, DecimalEscape, mev)
+    
+    //let private decimalEscapesToHexEscapes str =
+    //    let mev = MatchEvaluator (fun m -> decimalEscapeToHexEscape m.Value) 
+    //    Regex.Replace(str, DecimalEscape, mev)
+
+    //let private unicodeEscapesToUTFEscapes str = 
+    //    let mev = MatchEvaluator (fun m -> unicodeEscapeToUEscape m.Value) 
+    //    Regex.Replace(str, UnicodeEscape, mev)
+    
+
+    ///
+
+    let escapeToString (esc: string) : string = 
+           Regex.Unescape esc
+
+    let decimalEscapeToString (decEsc : string) : string = 
+        decimalEscapeToUtf32 decEsc
+        |> Char.ConvertFromUtf32
+
+    let hexEscapeToString (hexEsc : string) : string =
+        hexEscapeToUtf32 hexEsc
+        |> Char.ConvertFromUtf32
+
+    let unicodeEscapeToString (uniEsc : string) : string =
+        unicodeEscapeToUtf32 uniEsc
+        |> Char.ConvertFromUtf32
+
+    /// This function replaces any end of line sequence by linefeed '\n'.
+    let normalizeEndOfLine str =
+        Regex.Replace(str, EndOfLine, Linefeed)
+
+    /// This function removes line continuation in strings. 
+    /// It assumes, that end of line is normalized to line feed
+    let replaceLineContinuations blechString = 
+        Regex.Replace(blechString, LineContinuation, "")
+
+    let replaceEscapes blechString = 
+        let mev = MatchEvaluator (fun m -> escapeToString m.Value) 
+        Regex.Replace(blechString, Escape, mev)
+    
+    let replaceDecimalEscapes blechString = 
+        let mev = MatchEvaluator (fun m -> decimalEscapeToString m.Value) 
+        Regex.Replace(blechString, DecimalEscape, mev)
+        
+    let replaceHexEscapes blechString = 
+        let mev = MatchEvaluator (fun m -> hexEscapeToString m.Value) 
+        Regex.Replace(blechString, HexEscape, mev)
+    
+    let replaceUnicodeEscapes blechString = 
+        let mev = MatchEvaluator (fun m -> unicodeEscapeToString m.Value) 
+        Regex.Replace(blechString, HexEscape, mev)
+    
+    /// Normalize a string literal from the lexer
+
+
+    let unescapeStringLiteral blechString =
```

Comment:
Assuming the reason `Regex.Unescape` could not be used here was because it fails on non-printable escape sequences such as `\w`, please document this in the code.